src/WebJobs.Script.WebHost/Startup.cs

```diff
@@ -43,6 +35,13 @@ public void Configure(IApplicationBuilder app, IApplicationLifetime applicationL
                 app.UseDeveloperExceptionPage();
             }
 
+            var environment = app.ApplicationServices.GetService<IEnvironment>();
+            if (environment.IsLinuxContainerEnvironment())
+            {
+                // Linux containers always start out in placeholder mode
+                environment.SetEnvironmentVariable(EnvironmentSettingNames.AzureWebsitePlaceholderMode, "1");
+            }
```

Comment:
Moved this check down to remove dependencies on SystemEnvironment and ScriptSettingsManager. Still runs very early in the pipeline, but makes it more testable.